src/Sarif.Converters/CppCheckError.cs

```diff
@@ -132,49 +123,49 @@ public Result ToSarifIssue()
                 result.Message = new Message { Text = this.Message };
             }
 
-            PhysicalLocation lastLocationConverted;
-            if (this.Locations.Length == 1)
+            if (this.Locations.Length > 0)
             {
-                lastLocationConverted = this.Locations[0].ToSarifPhysicalLocation();
+                AddLocationToResult(this.Locations, result);
             }
-            else
+
+            return result;
+        }
+
+        private static void AddLocationToResult(IEnumerable<CppCheckLocation> cppCheckLocations, Result result)
+        {
+            PhysicalLocation lastLocationConverted;
+            var locations = new List<ThreadFlowLocation>
             {
-                var locations = new List<ThreadFlowLocation>
-                {
-                    Capacity = this.Locations.Length
-                };
+                Capacity = cppCheckLocations.Count()
+            };
 
-                foreach (CppCheckLocation loc in this.Locations)
+            foreach (CppCheckLocation loc in cppCheckLocations)
+            {
+                locations.Add(new ThreadFlowLocation
                 {
-                    locations.Add(new ThreadFlowLocation
+                    Location = new Location
                     {
-                        Location = new Location
-                        {
-                            PhysicalLocation = loc.ToSarifPhysicalLocation()
-                        },
-                        Importance = ThreadFlowLocationImportance.Essential
-                    });
-                }
-
-                result.CodeFlows = new List<CodeFlow>()
-                {
-                    SarifUtilities.CreateSingleThreadedCodeFlow(locations)
-                };
-
-                // In the N != 1 case, set the overall location's location to
-                // the last entry in the execution flow.
-                lastLocationConverted = locations[locations.Count - 1].Location.PhysicalLocation;
+                        PhysicalLocation = loc.ToSarifPhysicalLocation()
+                    },
+                    Importance = ThreadFlowLocationImportance.Essential
+                });
             }
 
+            result.CodeFlows = new List<CodeFlow>()
+            {
+                SarifUtilities.CreateSingleThreadedCodeFlow(locations)
+            };
+
+            // In the N != 1 case, set the overall location's location to
```

Comment:
>In the N != 1 case, s [](start = 15, length = 21)

Let's fix this existing comment:

- You do this in call cases now, not just N != 1.
- It's the result's location not the "overall location's location".
- The SARIF term is "code flow", not "execution flow".

So all together, change the comment to just:

// Set the result's location to the last location in the code flow.